src/main/java/net/greatstart/controllers/InvestmentInterestController.java

```diff
@@ -0,0 +1,88 @@
+package net.greatstart.controllers;
+
+import net.greatstart.model.InvestmentInterest;
+import net.greatstart.model.User;
+import net.greatstart.services.InvestmentInterestService;
+import net.greatstart.services.UserService;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Controller;
+import org.springframework.validation.Errors;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestMethod;
+import org.springframework.web.servlet.ModelAndView;
+
+import javax.validation.Valid;
+import java.security.Principal;
+import java.util.List;
+
+@Controller
+public class InvestmentInterestController {
+    private static final String REDIRECT_TO_INVESTMENT_INTEREST = "redirect:/invinterest/";
+
+    private InvestmentInterestService investmentInterestService;
+    private UserService userService;
+
+    @Autowired
+    public InvestmentInterestController(InvestmentInterestService investmentInterestService, UserService userService) {
+        this.investmentInterestService = investmentInterestService;
+        this.userService = userService;
+    }
+
+    @RequestMapping("/invinterest")
+    public ModelAndView showInvestmentsInterest() {
+        List<InvestmentInterest> investmentInterestList = this.investmentInterestService
+                .getAllInvestmentInterest();
+        ModelAndView model = new ModelAndView("invinterest/invinterests");
+        model.addObject("inv_interest_list", investmentInterestList);
+        return model;
+    }
+
+    @RequestMapping(value = "/invinterest/add", method = RequestMethod.GET)
+    public ModelAndView getInvestmentInterestForm() {
+        ModelAndView model = new ModelAndView("invinterest/add_invinterest");
+        model.addObject("invinterest", new InvestmentInterest());
+        return model;
+    }
+
+    @RequestMapping(value = "/invinterest/add", method = RequestMethod.POST)
+    public ModelAndView addInvestmentInterest(InvestmentInterest investmentInterest,
+                                              Errors errors,
+                                              Principal principal) {
+        if (errors.hasErrors()) {
+            return new ModelAndView("invinterest/add_invinterest");
+        }
+        // TODO: get authenticated user once user authorization is ready

```

Comment:
![INFO](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-info.png 'Severity: INFO') Complete the task associated to this TODO comment. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS1135)